packages/replay-next/src/suspense/ExecutionPointsCache.ts

```diff
@@ -293,12 +294,24 @@ export function preCacheExecutionPointForTime(timeStampedPoint: TimeStampedPoint
   }
 }
 
+export const sessionEndPointCache = createSingleEntryCache<
+  [replayClient: ReplayClientInterface],
+  TimeStampedPoint
+>({
+  config: { immutable: true },
+  debugLabel: "SessionEndPointCache",
+  load: ([client]) => client.getSessionEndpoint(),
+});
+
 export const pointsBoundingTimeCache: Cache<
   [replayClient: ReplayClientInterface, time: number],
   PointsBoundingTime
 > = createCache({
   config: { immutable: true },
   debugLabel: "PointsBoundingTime",
   getKey: ([client, time]) => `${time}`,
-  load: async ([client, time]) => client.getPointsBoundingTime(time),
+  load: async ([client, time]) => {
+    const endpoint = await sessionEndPointCache.readAsync(client);
+    return client.getPointsBoundingTime(clamp(time, 0, endpoint.time));
```

Comment:
I guess I'm not necessarily opposed to the `sessionEndPointCache`– I wonder if the Replay client shouldn't be the thing enforcing this– since we could still call client methods with potentially invalid times in other ways than just this cache?

Reply:
Personally I prefer `ReplayClient` to be a thin wrapper around the protocol and not add any logic to it.

Reply:
It already has logic for e.g. waiting for a range to be loaded before making a request. This doesn't seem substantially different from that, does it?

Reply:
Btw I don't feel strongly enough about this to block the PR or anything 😄  It's just a preference.